jest/private/bazel_snapshot_resolver.js

```diff
@@ -37,24 +43,54 @@ let snapshotResolver = process.env.JEST_TEST__USER_SNAPSHOT_RESOLVER
   : defaultSnapshotResolver;
 
 if (process.env.JEST_TEST__UPDATE_SNAPSHOTS) {
-  // If we're updating snapshots then append a GENERATED_SNAPSHOT_SUFFIX to the snapshot path so the
-  // generated snapshot path is different from the source snapshot path allowing us to use
-  // write_source_files to for the update target.
-  origSnapshotResolver = snapshotResolver;
-  snapshotResolver = {
-    resolveSnapshotPath: (testPath) =>
-      origSnapshotResolver.resolveSnapshotPath(testPath) +
-      GENERATED_SNAPSHOT_SUFFIX,
-    resolveTestPath: (snapshotPath) =>
-      origSnapshotResolver.resolveTestPath(
-        snapshotPath.substr(
-          0,
-          snapshotPath.length - GENERATED_SNAPSHOT_SUFFIX.length
-        )
-      ),
-    testPathForConsistencyCheck:
-      origSnapshotResolver.testPathForConsistencyCheck,
-  };
+  // This run of Jest is meant for generating reference snapshots for the snapshots update Bazel target
+  if (process.env.JEST_TEST__UPDATE_SNAPSHOTS == "directory") {
+    // If we're updating snapshots with Bazel then switch the directory that snapshots are written
+    // to to REFERENCE_SNAPSHOT_DIRECTORY so the generated snapshot path is different from the
+    // source snapshot path allowing us to use write_source_files to for the update target.
+    origSnapshotResolver = snapshotResolver;
+    snapshotResolver = {
```

Comment:
I know this already existed below but I'd always vote to not assume which properties exist on the object and spread the original before overwriting some of it:
```
snapshotResolver = {
    ...origSnapshotResolver,

    resolveSnapshotPath: () { ... origSnapshotResolver.resolveSnapshotPath() ... }
}
```

Reply:
Yeah. That's really nice. 🧙